src/game/music.c

```diff
@@ -49,9 +56,7 @@ static void Music_StreamFinished(int stream_id, void *user_data)
 
     if (stream_id == m_AudioStreamID) {
         m_AudioStreamID = -1;
-        if (m_Track == MX_SECRET) {
```

Comment:
If you disable `fix_secrets_killing_music` then collect the first secret in Caves, the secret track plays, but if you then immediately go for the second secret, it doesn't play. Is it maybe because of this particular change?
The third secret is fine as there is other music triggered in between. And equally, it's all fine with this setting enabled as standard.

Reply:
Ah shoot this is a repeat of #311. The problem is it skips the secret track because the game intentionally doesn't play the same track twice in a row (which doesn't affect sound effects). I thought `m_TrackCurrent = MX_INACTIVE;` would be fine, but I forgot we added a `Music_Play` check for `track == m_TrackLastPlayed` also.

Reply:
I think we might need to add back in something like:
```
if (m_Track == MX_SECRET) {
    m_TrackLastPlayed = MX_INACTIVE;
}
```